### PR TITLE
Make SocksException more informative

### LINKED ISSUE #109404: Make SocksException(SR.net_socks_connection_failed) more informative
When it's impossible to connect socks5 proxy server dotnet [throws](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocksHelper.cs#L193) SocksException with general [message](https://github.com/dotnet/runtime/blob/d2a2a79b2eaf66d68a02f35917907bd9fbe97ea7/src/libraries/System.Net.Http/src/Resources/Strings.resx#L546) "SOCKS server failed to connect to the destination.". And it's hard to understand what exactly went wrong, because proxy server can return different "errors" bytes (from 0x01 to 0x08).

For example, golang throws/returns different errors for each different byte: https://github.com/golang/net/blob/511cc3a40645a2f6ed3d21a1d0803b5057b9aaa1/internal/socks/client.go#L115
https://github.com/golang/net/blob/511cc3a40645a2f6ed3d21a1d0803b5057b9aaa1/internal/socks/socks.go#L40-L63

So my proposal is include "error" byte (0x01-0x08) in SocksException or make different messages for each byte (how it's done in golang).

### PR DESCRIPTION
**Pull request by alexeev-igor**
Pass SOCKS server error code to SocksException when can not connect to SOCKS server

Fix #109404

### CODE REVIEW

**Comment by alexeev-igor** [comment]
@dotnet-policy-service agree

File: src/libraries/System.Net.Http/src/Resources/Strings.resx [hunk 1]
```diff
@@ -544,7 +544,7 @@
     <value>SOCKS server returned an unknown address type.</value>
   </data>
   <data name="net_socks_connection_failed" xml:space="preserve">
-    <value>SOCKS server failed to connect to the destination.</value>
+    <value>SOCKS server failed to connect to the destination. Received '{0}' from SOCKS server.</value>
```
Comment (MihaZupan) [inline src/libraries/System.Net.Http/src/Resources/Strings.resx:547]:
```suggestion
    <value>SOCKS server failed to connect to the destination. Received error code 0x{0}.</value>
```

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocksHelper.cs [hunk 2]
```diff
@@ -192,7 +192,7 @@ private static async ValueTask EstablishSocks5TunnelAsync(Stream stream, string
                 VerifyProtocolVersion(ProtocolVersion5, buffer[0]);
                 if (buffer[1] != Socks5_Success)
                 {
-                    throw new SocksException(SR.net_socks_connection_failed);
+                    throw new SocksException(SR.Format(SR.net_socks_connection_failed, buffer[1].ToString("00")));
```
Comment (MihaZupan) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocksHelper.cs:195]:
```suggestion
                    throw new SocksException(SR.Format(SR.net_socks_connection_failed, buffer[1].ToString("X2")));
```

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocksHelper.cs [hunk 3]
```diff
@@ -307,7 +307,7 @@ private static async ValueTask EstablishSocks4TunnelAsync(Stream stream, bool is
                     case Socks4_AuthFailed:
                         throw new SocksException(SR.net_socks_auth_failed);
                     default:
-                        throw new SocksException(SR.net_socks_connection_failed);
+                        throw new SocksException(SR.Format(SR.net_socks_connection_failed, buffer[1].ToString("00")));
```
Comment (MihaZupan) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocksHelper.cs:310]:
```suggestion
                        throw new SocksException(SR.Format(SR.net_socks_connection_failed, buffer[1].ToString("X2")));
```

**Review by MihaZupan — APPROVED** [review]
Thanks

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocksHelper.cs (hunk 2 above)
Comment (alexeev-igor) [reply src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocksHelper.cs]: Thanks, fixed.

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocksHelper.cs (hunk 3 above)
Comment (alexeev-igor) [reply src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocksHelper.cs]: Thanks, fixed.

**Pull request merged**